### DISCUSSION TITLE
12 Step Meeting Plugin Changes Address

### DISCUSSION THREAD

**Discussion by mwa1007** [code4recovery/12-step-meeting-list#1527] [General]
When I import the .csv, there is a meeting on Sunday @ 5:00pm with the address of 190 N Mail, Parowan, UT 84761. It imports fine but when I look at it on the website, it changes the zip code to 85760 and the city to Paragonah. Even if I go to the meeting plugin in word press and try to change the address manually, it changes it back. Please help, it’s mapping an address in a completely different city. Thanks,

Mike

**Comment by joshreisner** [comment]
we use google to confirm addresses, and looks like in this case it's putting the pin in the correct place, but mistaking the city name and postal code with a nearby city

<img width="1078" alt="Screenshot 2024-09-24 at 2 57 43 PM" src="https://github.com/user-attachments/assets/abb41cd9-c771-4a8e-8eb2-fc36b239ef35">

you should be able to fix this by adding this code to your theme's functions.php:

```
if (function_exists('tsml_custom_addresses')) {
    tsml_custom_addresses(array(
        '190 N Main St, Parowan, UT 84761, USA' => array(
            'formatted_address' => '190 N Main St, Parowan, UT 84761, USA',
            'city' => 'Parowan',
            'latitude' => 37.8453719,
            'longitude' => -112.8273439,
            'approximate' => 'no',
        ),
        '190 N Main St, Paragonah, UT 84760' => array(
            'formatted_address' => '190 N Main St, Parowan, UT 84761, USA',
            'city' => 'Parowan',
            'latitude' => 37.8453719,
            'longitude' => -112.8273439,
            'approximate' => 'no',
        ),
    ));
}
```

**Comment by mwa1007** [comment]
Josh,

Thanks so much for the info. I’m a bit new to Wordpress and website management in general. Where is the theme’s functions.php that needs to be edited?

From: Josh Reisner ***@***.***>
Date: Tuesday, September 24, 2024 at 4:06 PM
To: code4recovery/12-step-meeting-list ***@***.***>
Cc: mwa1007 ***@***.***>, Author ***@***.***>
Subject: Re: [code4recovery/12-step-meeting-list] 12 Step Meeting Plugin Changes Address (Discussion #1527)

we use google to confirm addresses, and looks like in this case it's putting the pin in the correct place, but mistaking the city name and postal code with a nearby city
Screenshot.2024-09-24.at.2.57.43.PM.png (view on web)<https://github.com/user-attachments/assets/abb41cd9-c771-4a8e-8eb2-fc36b239ef35>

you should be able to fix this by adding this code to your theme's functions.php:

if (function_exists('tsml_custom_addresses')) {

    tsml_custom_addresses(array(

        '190 N Main St, Parowan, UT 84761, USA' => array(

            'formatted_address' => '190 N Main St, Parowan, UT 84761, USA',

            'city' => 'Parowan',

            'latitude' => 37.8453719,

            'longitude' => -112.8273439,

            'approximate' => 'no',

        ),

        '190 N Main St, Paragonah, UT 84760' => array(

            'formatted_address' => '190 N Main St, Parowan, UT 84761, USA',

            'city' => 'Parowan',

            'latitude' => 37.8453719,

            'longitude' => -112.8273439,

            'approximate' => 'no',

        ),

    ));

}

**Reply by joshreisner** [reply]
if you google it there are a bunch of links like this: https://jetpack.com/blog/wordpress-functions-php/

i'd recommend making this change either through your hosting control panel or through FTP, if you have that. editing it through wordpress historically has been a little risky, although i'm not sure that's true anymore

if your theme is called `mytheme` then the path should be `/wp-content/themes/mytheme/functions.php`

typically this type of code is added at the bottom of the file

**Comment by mwa1007** [comment]
Josh,Thanks so much for your help. I’ll give it a try through FTP. With a backup of course. Regards,Mike Anderson(704) 877-4516On Sep 24, 2024, at 7:16 PM, Josh Reisner ***@***.***> wrote:﻿
if you google it there are a bunch of links like this: https://jetpack.com/blog/wordpress-functions-php/
i'd recommend making this change either through your hosting control panel or through FTP, if you have that. editing it through wordpress historically has been a little risky, although i'm not sure that's true anymore
if your theme is called mytheme then the path should be /wp-content/themes/mytheme/functions.php
typically this type of code is added at the bottom of the file
